tensorboard/plugins/hparams/backend_context.py

```diff
@@ -166,18 +169,18 @@ def _find_experiment_tag(self, experiment_id):
         Returns:
           The experiment or None if no such experiment is found.
         """
-        with self._experiment_from_tag_lock:
-            experiment = self._experiment_from_tag.get(experiment_id)
-            if experiment is None:
-                mapping = self.hparams_metadata(experiment_id)
-                for tag_to_content in mapping.values():
-                    if metadata.EXPERIMENT_TAG in tag_to_content:
-                        experiment = metadata.parse_experiment_plugin_data(
-                            tag_to_content[metadata.EXPERIMENT_TAG]
-                        )
-                        self._experiment_from_tag[experiment_id] = experiment
-                        break
-        return experiment
+        mapping = self.hparams_metadata(
+            experiment_id,
+            run_tag_filter=provider.RunTagFilter(
+                tags=[metadata.EXPERIMENT_TAG]
+            ),
+        )
+        if not mapping:
+            return None
+        # We expect only one experiment; pick arbitrarily.
+        tag_to_content = mapping[next(iter(mapping))]
```

Comment:
nit: `next(iter(mapping.values()))` seems slightly simpler than feeding the key back into the dict?  Same below